lib/consul/index.js

```diff
@@ -89,41 +90,94 @@ var populate_kvs_from_json = function(branch, prefix, obj, cb) {
-    file_deleted(branch, file, function(err) {
-      if (err) return cb('Failed to delete key ' + key_name + ' due to ' + err);
-
-      fs.readFile(fqf, {encoding:'utf8'}, function(err, body) {
+  var handle_json_kv_file = function(file_path, cb) {
+      fs.readFile(file_path, {encoding: 'utf8'}, function (err, body) {
         /* istanbul ignore if */
-        if (err) return cb('Failed to read key ' + fqf + ' due to ' + err);
+        if (err) return cb('Failed to read key ' + file_path + ' due to ' + err);
         var body = body ? body.trim() : '';
         try {
           var obj = JSON.parse(body);
-          populate_kvs_from_json(branch, create_key_name(branch, file), obj, cb);
-        } catch(e) {
+          populate_kvs_from_object(branch, create_key_name(branch, file), obj, cb);
+        } catch (e) {
           logger.warn("Failed to parse .json file.  Using body string as a KV.");
           write_content_to_consul(create_key_name(branch, file), body, cb);
         }
       });
+  };
+
+  var handle_properties_kv_file = function(file_path, common_properties_relative_path, cb) {
+
+    function extract_and_populate_properties(file_body, common_body, cb) {
+      utils.load_properties(file_body, common_body, function (error, obj) {
+        if (error) {
+          logger.warn('Failed to load properties for : ' + file + ' due to : ' + error + '.' + ' Using body string as a KV.');
+          handle_as_flat_file(file_path, branch, file, cb);
+        } else {
+          populate_kvs_from_object(branch, create_key_name(branch, file), obj, cb);
+        }
+      });
+    }
+
+    fs.readFile(file_path, {encoding: 'utf8'}, function (err, file_body) {
+      /* istanbul ignore if */
+      if (err) return cb('Failed to read key ' + file_path + ' due to ' + err);
+
+      if(common_properties_relative_path) {
+        var path_to_common_properties_file = branch.branch_directory + path.sep + common_properties_relative_path;
+        fs.readFile(path_to_common_properties_file, {encoding: 'utf8'}, function (err, common_body) {
+          if (err) {
+            logger.warn('Failed to read common variables for ' + path_to_common_properties_file + ' due to ' + err);
+            common_body = '';
+          }
+          extract_and_populate_properties(file_body, common_body, cb);
+        });
+      }
+      else {
+        extract_and_populate_properties(file_body, '', cb);
+      }
     });
-  } else {
-    fs.readFile(fqf, {encoding:'utf8'}, function(err, body) {
+  };
+
+  var handle_as_flat_file = function(fqf, branch, file, cb) {
+    fs.readFile(fqf, {encoding: 'utf8'}, function (err, body) {
       /* istanbul ignore if */
       if (err) return cb('Failed to read key ' + fqf + ' due to ' + err);
       var body = body ? body.trim() : '';
       write_content_to_consul(create_key_name(branch, file), body, cb);
     });
+  };
+
+  var handle_expended_keys_with_different_file_types = function() {
```

Comment:
Spelling issue - functionally fine, but I believe this function should be named "_expanded_" and not "_expended_"